pkg/build/webhook/generic/generic.go

```diff
@@ -65,8 +66,15 @@ func (p *WebHookPlugin) Extract(buildCfg *api.BuildConfig, secret, path string,
 		}
 
 		var data api.GenericWebHookEvent
+		if contentType == "application/yaml" {
+			body, err = yaml.ToJSON(body)
+			if err != nil {
+				glog.V(4).Infof("Error converting payload to json %v, but continuing with build", err)
+				return nil, envvars, true, nil
+			}
+		}
 		if err = json.Unmarshal(body, &data); err != nil {
-			glog.V(4).Infof("Error unmarshaling json %v, but continuing", err)
+			glog.V(4).Infof("Error unmarshalling payload %v, but continuing with build", err)
```

Comment:
i don't remember exactly why we made that choice, but i think we're mostly stuck with it now.